### DISCUSSION TITLE
Can you blur specific elements of a LayerSurface?

### DISCUSSION THREAD

**Discussion by colin-heffernan** [hyprwm/Hyprland#748] [Q&A] [answered]
I'm using EWW and I want to be able to blur certain widgets, but not others. Is this possible? I asked this in EWW's repo and they said that blurring was the compositor's job.

**Comment by avalsch** [comment, marked as answer]
Probably not, since eww uses gtk-layer-shell, which is one surface to hyprland, so there's no way to tell hyprland where to blur.

**Comment by fufexan** [comment]
I think it's possible, if you have multiple `defwindow`s.

```
› hyprctl layers
Monitor eDP-1:
	Layer level 0:
		Layer 1dd0bd0: xywh: 0 0 1598 999, namespace: wallpaper
	Layer level 1:
	Layer level 2:
		Layer 1db2d50: xywh: 0 0 1598 32, namespace: gtk-layer-shell
		Layer 1450830: xywh: 1597 42 1 946, namespace: gtk-layer-shell
	Layer level 3:
```
This is how it looks on my setup. `1db2d50` is my bar and `1450830` is my notification panel.

You can blur surfaces using `blurls`: https://wiki.hyprland.org/Configuring/Keywords/#blurring-layersurfaces.

**Reply by avalsch** [reply]
How would you distinguish between different widgets if they have the same namespace though?

**Reply by colin-heffernan** [reply]
Wouldn't this not work because all of them share the same namespace?

**Reply by colin-heffernan** [reply]
Also, I tested what happens after I open and close the widgets. The layer ID changes.

**Reply by fufexan** [reply]
Ah yeah. You're right. My mind was elsewhere.

Then I guess no, you can't do that unless eww can display different namespaces to different widgets (if at all possible).

**Reply by colin-heffernan** [reply]
For context on why I need to be able to open and close the widgets: I'm trying to make a start menu with a blurred background. The issue is that the widgets have rounded corners, and the parts where the corners are transparent are blurred.

**Reply by colin-heffernan** [reply]
Alright, then I guess I'll go without blur. Thanks for the help.

**Reply by fufexan** [reply]
Yeah, hyprland can't really know how to cut the blur on corners, since all layersurfaces report is a rectangle.

**Reply by colin-heffernan** [reply]
I opened an issue in EWW as a feature suggestion allowing for different namespaces on different defwindows. Hopefully it's possible to implement.
